src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -176,7 +182,12 @@ protected ExtensionsRunner(Extension extension) throws IOException {
             .put(TransportSettings.BIND_HOST.getKey(), extensionSettings.getHostAddress())
             .put(TransportSettings.PORT.getKey(), extensionSettings.getHostPort())
             .build();
-        this.threadPool = new ThreadPool(settings);
+
+        final List<ExecutorBuilder<?>> executorBuilders = extension.getExecutorBuilders(settings);
+
+        runnableTaskListener = new AtomicReference<>();
```

Comment:
(Style Preference) I really like to use `this.foo` in Constructors when setting final fields.

Reply:
@joshpalis I think you misunderstood the comment.  The executorBuilders is fine as a local variable. I meant to add this to the runnableTaskListener.

Reply:
Oh I misread, let me go ahead and fix this :)